### PR TITLE
Display Case Specific Error Messages in Home Realm Discovery if Enabled

### PR DESCRIPTION
**Pull request by dhaura**
### Purpose
- $Subject

### Related Issues
- https://github.com/wso2/product-is/issues/18896

### Related PRs
- https://github.com/wso2/carbon-identity-framework/pull/5415

### CODE REVIEW

File: identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources.properties [hunk 1]
```diff
@@ -97,6 +97,7 @@ select.tenant.dropdown.display.name=Select Tenant
 super.tenant.display.name=Super Tenant
 super.tenant=carbon.super
 domain.unknown=domain.unknown
+idp.not.assigned.to.sp=idp.not.assigned.to.sp
```
Comment (pavinduLakshan) [inline identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources.properties:100]: Is using key for the value here intentional? 

Check the other places as well.

File: identity-apps-core/apps/authentication-portal/src/main/webapp/domain.jsp [hunk 2]
```diff
@@ -37,6 +38,8 @@
 
             if (domainUnknown.equalsIgnoreCase(errorMessage)) {
                 errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "domain.cannot.be.identified");
+            } else if (idpNotAssignedToSP.equalsIgnoreCase(errorMessage)) {
```
Comment (pavinduLakshan) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/domain.jsp:41]: There's no need to modify translations for this key. Just directly comparing the i18n key with errorMessage is enough. 

Ref: 
https://github.com/wso2/identity-apps/blob/master/identity-apps-core/apps/authentication-portal/src/main/webapp/org_name.jsp#L49

Comment (dhaura) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/domain.jsp]: Resolved with https://github.com/wso2/identity-apps/pull/5322/commits/7891a913a44c8c6f267ed417e5b3c4ce633dc7a3

File: identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources.properties [hunk 3]
```diff
@@ -116,6 +116,7 @@ username.or.password.invalid=Username or Password is Invalid
 create.an.account=Create an account
 unauthorized.to.login=You are not authorized to login
 domain.cannot.be.identified=Domain cannot be identified! Please retry.
+domain.cannot.be.identified.for.sp=Domain cannot be identified for the service provider! Please retry.
```
Comment (pavinduLakshan) [inline identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources.properties:119]: Is the use of `!` here intentional?

**Comment by dhaura** [comment]
Closing this PR since the proposed solution is under reconsideration.

**Pull request closed without merging**